lib/bats-core/tracing.bash

```diff
@@ -23,6 +23,15 @@ bats_capture_stack_trace() {
 	done
 }
 
+bats_get_failure_stack_trace() {
+	# See bats_debug_trap for details.
+	if [[ -n "${BATS_DEBUG_LAST_STACK_TRACE_IS_VALID}" ]]; then
+		eval "${1}"'=("${BATS_DEBUG_LAST_STACK_TRACE[@]}")'
```

Comment:
You may need the same `${arr[@]+"${arr[@]}"}` expansion as below (maybe not, just calling it out).